pallets/gear/src/benchmarking/mod.rs

```diff
@@ -503,39 +537,56 @@ benchmarks! {
         let origin = RawOrigin::Signed(caller);
 
         init_block::<T>(None);
-    }: _(origin, code, salt, vec![], 100_000_000_u64, value)
+    }: _(origin, code, salt, vec![], GAS_LIMIT_EXT, value)
     verify {
         assert!(matches!(QueueOf::<T>::dequeue(), Ok(Some(_))));
     }
 
-    send_message {
+    send_message_program_interaction {
         let p in 0 .. MAX_PAYLOAD_LEN;
         let caller = benchmarking::account("caller", 0, 0);
-        <T as pallet::Config>::Currency::deposit_creating(&caller, 100_000_000_000_000_u128.unique_saturated_into());
+        <T as pallet::Config>::Currency::deposit_creating(&caller, DEPOSIT_AMOUNT.unique_saturated_into());
         let minimum_balance = <T as pallet::Config>::Currency::minimum_balance();
-        let program_id = ProgramId::from_origin(benchmarking::account::<T::AccountId>("program", 0, 100).into_origin());
+        let program_id = ProgramId::from_origin(benchmarking::account::<T::AccountId>("program", 0, PROGRAM_ID_DEFAULT).into_origin());
         let code = benchmarking::generate_wasm2(16.into()).unwrap();
         benchmarking::set_program::<ProgramStorageOf::<T>, _>(program_id, code, 1.into());
         let payload = vec![0_u8; p as usize];
 
         init_block::<T>(None);
-    }: _(RawOrigin::Signed(caller), program_id, payload, 100_000_000_u64, minimum_balance)
+    }: send_message(RawOrigin::Signed(caller), program_id, payload, GAS_LIMIT_EXT, minimum_balance)
     verify {
         assert!(matches!(QueueOf::<T>::dequeue(), Ok(Some(_))));
     }
 
+    send_message_user_interaction {
+        let p in 0 .. MAX_PAYLOAD_LEN;
+        let caller = benchmarking::account("caller", 0, 0);
+        <T as pallet::Config>::Currency::deposit_creating(&caller, DEPOSIT_AMOUNT.unique_saturated_into());
+        let minimum_balance = <T as pallet::Config>::Currency::minimum_balance();
+        let user_id = ProgramId::from(12345);
+        let payload = vec![0_u8; p as usize];
+
+        init_block::<T>(None);
+    }: send_message(RawOrigin::Signed(caller.clone()), user_id, payload, GAS_LIMIT_EXT, minimum_balance)
+    verify {
+        assert!(matches!(QueueOf::<T>::dequeue(), Ok(None)));
+        assert!(MailboxOf::<T>::is_empty(&caller));
+        let user = T::AccountId::from_origin(user_id.into_origin());
+        assert!(MailboxOf::<T>::is_empty(&user));
```

Comment:
so where is the message, there should be event assertion

Reply:
I don't know how to do it properly 😄

Reply:
Go through events and assert  existence of `pallet_gear::Event::UserMessageSent { expiration: None, ... }`.
```rust
// How to query events within the benchmarks scope
let events = frame_system::Pallet::<T>::events();
```

Reply:
Done, please review.